desidiff/nb/desidiff.ipynb

```diff
@@ -196,15 +205,44 @@
     "            plt.show()\n",
     "            count=count+1\n",
     "            \n",
+    "            #SkyPortal functionality begins:\n",
+    "            #Combine the wavelengths into one \n",
+    "            wavelengths = difwave['b'].tolist() + difwave['r'].tolist() + difwave['z'].tolist()\n",
+    "            \n",
+    "           #Combine the differenced flux into one \n",
+    "            flux = difflux['b'].tolist() + difflux['r'].tolist() + difflux['z'].tolist()\n",
+    "            \n",
+    "            \n",
+    "            #SkyPortal's Id for the object\n",
+    "            objID = 'DESI{}'.format(str(tid[0]))\n",
+    "            \n",
+    "            #Code to check if this object already exists in SkyPortal. If not, create it\n",
+    "            response = requests.get(\"http://desi2.lbl.gov:5000/api/candidates/{}\".format(objID), headers={\"Authorization\": f\"token {token}\"})\n",
+    "            if response.status_code == 400:\n",
+    "                obj_data = {\n",
+    "                        \"ra\": 0, #RA is required when creating a new object, I have currently set them all to 0, since it doesn't affect our purpose of uploading the spectrum. We should discuss this. \n",
+    "                        \"dec\": 0, #Same for DEC\n",
+    "                        \"id\": objID,\n",
+    "                        \"filter_ids\": [sp.filter_id(filter_name)],\n",
+    "                        \"passed_at\": str(datetime.datetime.utcnow()) #UTC time when the object passed the filter\n",
+    "                        }\n",
+    "                sp.api('POST', '{}/api/candidates/obj_id'.format(sp.url), data = obj_data)\n",
+    "                \n",
+    "            #Now, we send the differenced spectrum to SkyPortal\n",
+    "            spectrum_data = {\n",
+    "                        \"obj_id\": objID,\n",
+    "                        \"wavelengths\": wavelengths,\n",
+    "                        \"fluxes\": flux,\n",
+    "                        \"observed_at\": str(datetime.datetime.utcnow()), #UTC time when the difference spectrum was created, same as filter-passing time\n",
```

Comment:
Give the referenced spectrum the observed_at the date of the new spectrum.  Not the current date.